enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,276 @@
+
+## Summary
+
+The proposal adds a new field `executorSubject` to the spec of `ManifestWork` 
+api, to support dynamic identity authentication/authorization upon the execution
+of `ManifestWork` resources. If the `executorSubject` is absent, it means the
+work agent will keep using the mounted service account to raise requests against
+the managed cluster which is also for preserving the backward compatibility.
+
+## Motivation
+
+### Separate the identity and its held permission for the work agent
+
+There can be different actors using the work api to distribute resource from
+the hub cluster. Currently the work agent has the cluster-admin access to the
+managed cluster which means that every hub users with write access to the
+work resources will be able to dispatch any resources to the managed cluster.
+This can be a security hole for OCM, e.g. one can apply/overwrite some
+cluster-scoped configuration to the managed cluster which potentially break/halt
+the cluster. Hence, we need a way to clarify the owner identity/role of the
+"ManifestWork" before it takes effect so that we can explicitly check whether 
+that owner has sufficient permission in the managed cluster.
+
+### Auditing the API request history from work agent
+
+Currently all the API requests raised by work agent in the managed cluster are
+from a fixed service account ("open-cluster-management-agent/klusterlet-work-sa"
+under the default installation). However in some cases, we need to clarify the
+source/owner identity of the `ManifestWork` resource so that we can record the
+api audits in a finer granularity. This will be helpful for the managed
+cluster's admin to be capable of knowing "who is operating that `ManifestWork`
+from the hub cluster".
+
+### Goals
+
+- `ManifestWork` resource be able to clarify the executor/owner identity before
+  actually writing the resources to the managed cluster.
+
+### Non-Goals
+
+- This proposal doesn't put any requirement to the selection of executor/owner in
+  the "ManifestWork", which allows hub admin to reference any valid Kubernetes 
+  user/role to be the executor/owner.
+  
+## Terms
+
+- `Executor Subject`: The subject which is used by the work agent when actually 
+  applying resources in the spoke cluster.
+
+
+## Function Spec
+
+1. A hub admin can limit the scope of the dispatch'able resources for a 
+   `ManifestWork` by setting a non-empty executor subject, so the work agent
+   in the spoke cluster can report failure when applying/modifying out-of-scope 
+   resources.
+2. A spoke admin can expand the permission scope of the executor subject 
+   (e.g. a service-account) by binding more ClusterRole/Role to it.
+3. (beta phase) A hub user should pass the authorization before writing
+   the resources in the work object. The hub cluster holds a policy configuration
+   of "which hub user has write access to which work resource" and a validating
```

Comment:
IIRC, this means check if "user A can create/update resource B", right? 
Is it also possible to check if "user A can create/update resource B in namespace C"?